src/operation.rs

```diff
@@ -1142,6 +1149,164 @@ impl ValidateDocument {
+    pub fn from_arg_matches(
+        matches: &ArgMatches,
+    ) -> Result<Self, Failed> {
+        Ok(
+            <Self as FromArgMatches>::from_arg_matches(
+                matches
+            ).unwrap()
+        )
+    }
+
+    /// Tries to validate a document through RTA signatures.
+    ///
+    /// Returns successfully if validation is successful or with an
+    /// appropriate error otherwise.
+    fn run(self, process: Process) -> Result<(), ExitError> {
+        let mut validation = Engine::new(process.config(), !self.noupdate)?;
+        validation.ignite()?;
+        process.switch_logging(false, false)?;
+
+        // Load and decode the signature.
+        let data = match fs::read(&self.signature) {
+            Ok(data) => Bytes::from(data),
+            Err(err) => {
+                error!(
+                    "Failed to read signature '{}': {}",
+                    self.signature.display(), err
+                );
+                return Err(ExitError::Generic)
+            }
+        };
+        let rsc = match Rsc::decode(data, process.config().strict) {
+            Ok(rsc) => rsc,
+            Err(err) => {
+                error!(
+                    "Failed to decode signature '{}': {}",
+                    self.signature.display(), err
+                );
+                return Err(ExitError::Invalid)
+            }
+        };
+        
+        let mut hashes = Vec::new();
+        for item in rsc.iter() {
+            hashes.push(item);
+        }
+
+        'outer: for document in &self.document {
+            let digest = match rsc.digest_algorithm().digest_file(document) {
+                Ok(digest) => digest,
+                Err(err) => {
+                    error!(
+                        "Failed to read document '{}': {}",
+                        document.display(), err
+                    );
+                    return Err(ExitError::Generic)
+                }
+            };
+            let Some(doc_file_name) = document.file_name() else {
+                error!(
+                    "Failed to read document file name '{}'",
+                    document.display()
+                );
+                return Err(ExitError::Generic)
+            };
+            let doc_file_name = doc_file_name.to_string_lossy();
+            
+            for item in hashes.iter() {
+                if item.hash() == digest.as_ref() {
+                    if let Some(file_name) = item.file_name() {
+                        if file_name == doc_file_name.as_bytes() {
+                            continue 'outer;
+                        }
+                    } else {
+                        continue 'outer;
+                    }
+                }
+            }
+
+            // At this point none of the entries in the check list matched the
+            // document we supplied.
+
+            error!("Failed to match document to valid entry in the check list '{}'.", 
```

Comment:
Perhaps collect all the failed document file names into a vec and print all of them if that vec doesn’t end up being empty.